common/json_compare.go

```diff
@@ -0,0 +1,111 @@
+package common
+
+// Compare the JSON values (intended to compare override json and config output
+// in case of SLZ but can be used anywhere)
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+// SortSlice sorts the slices recursively.
+// It also takes care of any nested slices or maps inside the slice.
+func SortSlice(slice []interface{}) {
+	for i, item := range slice {
+		switch item := item.(type) {
+		case []interface{}: // If the item is a slice, sort it
+			SortSlice(item)
+		case map[string]interface{}: // If the item is a map, sort it
+			SortMap(item)
+		}
+		slice[i] = item
+	}
+	// Sort the slice itself. Uses string representation for comparison.
+	sort.SliceStable(slice, func(i, j int) bool {
+		return fmt.Sprintf("%s", slice[i]) < fmt.Sprintf("%s", slice[j])
+	})
+}
+
+// SortMapKeys sorts the keys of a map and returns them as a slice.
+func SortMapKeys(m map[string]interface{}) []string {
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+// SortMap sorts the map and any nested structures inside it.
+// It modifies the original map.
+func SortMap(m map[string]interface{}) {
+	for key, val := range m {
+		switch val := val.(type) {
+		case []interface{}: // If the value is a slice, sort it
+			SortSlice(val)
+			m[key] = val
+		case map[string]interface{}: // If the value is a map, sort it
+			SortMap(val)
+		}
+	}
+	// Sort the map itself based on the keys
+	keys := SortMapKeys(m)
+	sortedMap := make(map[string]interface{})
+	for _, key := range keys {
+		sortedMap[key] = m[key]
+	}
+	// Update the original map with sorted key-value pairs
+	for k := range m {
+		delete(m, k)
+	}
+	for k, v := range sortedMap {
+		m[k] = v
+	}
+}
+
+func GetJsonComparison(jsonFile1 string, jsonFile2 string) bool {
```

Comment:
Lets change the verb of this so it makes sense with a boolean return, instead of "Get", maybe something like "IsJsonEqual"?. I'll think about it, not sure, but probably start with "Is" since boolean.

It is a bit dangerous to be consuming the errors inside the function, the calling procedure will not know if the actual compare was unequal, or there was an error? I would suggest you also return the error object, and do not "print" the errors, just return them. If you want to format the error text to include the more descriptive messages you have now, look at doing something similar to this, which returns an error object that you can return:
`fmt.Errorf("some extra text: %w", err)`
"err" is whatever your error variable is, the "%w" will extract the original error message from it, and golang wants you to always start your error messages with a lower-case letter (linter will catch this).

Reply:
Sure. Thank you for mentioning these best practices.
I have modified the function name now as "IsJsonEqual" and also removed the part where errors are printed and only returning the formatted errors using `Errorf`